BLUESPAWN-client/headers/util/permissions/permissions.h

```diff
@@ -212,6 +260,81 @@ namespace Permissions {
 		 * @return The name of the owner
 		 */
 		virtual std::wstring ToString() const;
+
+		/**
+		* Function to get a list of privileges the owner has
+		*
+		* @return the list of privileges an owner explicity has, and if they 
+		*     are a user, includes the privileges granted to them by the
+		*     groups they're in.
+		*/
+		std::vector<LSA_UNICODE_STRING> GetPrivileges();
+
+		/**
+		* Function to check if the owner has a certain privilege
+		*
+		* @param wPriv the name of the privilege to check for. Names should
+		*    be given as constants listed here: https://docs.microsoft.com/en-us/windows/win32/secauthz/privilege-constants
+		* 
+		* @return true if the owner has the privilege, false otherwise
+		*/
+		bool HasPrivilege(IN const std::wstring& wPriv);
+
+		/**
+		* Function to check if a privilege is contained in a list of privileges
+		*
+		* @param vPrivList a vector containing the list of privileges to search through
+		* @param wPriv a wstring containing the name of the privilege to check for
+		*     Names should be from: https://docs.microsoft.com/en-us/windows/win32/secauthz/privilege-constants
+		*
+		* @return true if the privilege is is the list, false otherwise
+		*/
+		static bool PrivListHasPrivilege(IN const std::vector<LSA_UNICODE_STRING>& vPrivList, IN const std::wstring& wPriv) ;
```

Comment:
this probably shouldn't be part of the Owner class. If you can change the GetPrivileges() function to return wstrings, then this shouldn't be needed at all